File: backend/src/main/java/top/vuhe/entity/equipment/dao/StatusDAO.java
package top.vuhe.entity.equipment.dao;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;
import top.vuhe.entity.BaseEntity;

/**
 * @author zhuhe
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName(value = "status_info")
public class StatusDAO extends BaseEntity {
    private String statusName;
    private Color statusColor;

    public enum Color {
        // 灰色
        Gray,
        // 绿色
        Green,
        // 黄色
        Yellow,
        // 红色
        Red
    }
}

File: backend-kt/src/main/kotlin/top/vuhe/sw/entity/equipment/dao/StatusDAO.kt
package top.vuhe.sw.entity.equipment.dao

import com.baomidou.mybatisplus.annotation.TableName
import top.vuhe.sw.entity.equipment.StatusColor

@TableName(value = "status_info")
data class StatusDAO(
    val id: Int?,
    val statusName: String?,
    val statusColor: StatusColor,
)

